cobalt/android/apk/app/src/main/java/dev/cobalt/coat/CobaltService.java

```diff
@@ -87,13 +94,16 @@ public void onClose() {
 
   /**
    * Send data from the service to the client.
-   *
    */
   protected void sendToClient(long nativeService, byte[] data) {
-    // TODO(b/372558900): Implement Javascript Injection
-  }
+    if (this.cobaltActivity == null) {
+      Log.e(TAG, "CobaltActivity is null, can not run evaluateJavaScript()");
+      return;
+    }
 
-  private void nativeSendToClient(long nativeService, byte[] data) {
-    // TODO(b/372558900): Implement Javascript Injection
+    // Use Base64.NO_WRAP instead of Base64.DEFAULT to avoid adding a new line.
+    String base64Data = Base64.encodeToString(data, Base64.NO_WRAP);
+    String jsCode = String.format("window.H5vccPlatformService.callback_from_android(%d, '%s');", nativeService, base64Data);
```

Comment:
if this line is demo code, do you want to remove this before submitting?

Reply:
No, this is the code for platform service.

Reply:
I think web API functions use camelCase.

example: https://source.corp.google.com/h/github/youtube/cobalt/+/25.lts.1+:cobalt/h5vcc/h5vcc_net_log.idl;drc=301a02e59ea4d7f4d81f1d13a11b3bd26f83f7a8;l=18